tests/Form/AbstractLayoutTestCase.php

```diff
@@ -98,7 +101,7 @@ final protected function assertMatchesXpath(string $html, string $expression, in
                 1 === $count ? 'once' : $count.' times',
                 1 === $nodeList->length ? 'once' : $nodeList->length.' times',
                 // strip away <root> and </root>
-                substr($dom->saveHTML(), 6, -8)
+                substr($dom->saveHTML() ?: '', 6, -8)
```

Comment:
Couldn't be `0` a valid value?

Reply:
I wanted to handle the `false` value ; any recommendation ?

Reply:
I would say that `0` is not a valid value since it's HTML.

It's always supposed to be`<root>...</root>` since we're stripping away the root tag.

Reply:
>I wanted to handle the false value ; any recommendation ?
```php
substr(false !== $dom->saveHTML() ? $dom->saveHTML() : '', 6, -8)
```

Reply:
Since we're doing substr, it doesn't matter
```
substr('0', 6, -8) === false === substr('', 6, -8)
```